packages/shared_preferences/shared_preferences_android/lib/src/shared_preferences_android.dart

```diff
@@ -39,20 +41,27 @@ class SharedPreferencesAndroid extends SharedPreferencesStorePlatform {
   Future<bool> setValue(String valueType, String key, Object value) async {
     switch (valueType) {
       case 'String':
-        return _api.setString(key, value as String);
+        value as String;
+        if (value.startsWith(listPrefix) ||
+            value.startsWith(jsonListPrefix) ||
+            value.startsWith(doublePrefix)) {
+          throw ArgumentError(
+              'The string $value cannot be stored as it clashes with special identifier prefixes');
+        }
+        return _api.setString(key, value);
       case 'Bool':
         return _api.setBool(key, value as bool);
       case 'Int':
         return _api.setInt(key, value as int);
       case 'Double':
         return _api.setDouble(key, value as double);
       case 'StringList':
+        return _api.setString(key, '$jsonListPrefix${jsonEncode(value)}');
+      case 'LegacyStringListForTesting':
```

Comment:
https://github.com/flutter/flutter/blob/master/docs/contributing/Style-guide-for-Flutter-repo.md#avoid-newold-modifiers-in-code